### PR TITLE
Fix rollback primary collisions

### PR DESCRIPTION
**Pull request by JohnAFernandez**
Taylor, Wookiejedi, and I have all been going back and forth heavily testing these fixes.  On stationary targets, we have 100% accuracy.  On moving targets, we have one more test pending, but results are promising so far, even at high pings.  This improvement works best with Interpolation Overhaul 2, but is independent of it.

### CODE REVIEW

**Comment by JohnAFernandez** [comment]
Moving this back to draft for a little bit, because I just realized there is one more fix missing for doubled collisions.

File: code/network/multi_obj.cpp [hunk 1]
```diff
@@ -703,19 +731,28 @@ void multi_ship_record_do_rollback()
...
 
+	bool continue_rollback = true;
+
 	do {
 		// move all ships to their recorded positions
 		multi_oo_restore_frame(frame_idx);
 
 		// push weapons forward for the frame (weapons do not get pushed forward for the first frame of their existence)
-		multi_oo_simulate_rollback_shots(frame_idx);
+		continue_rollback = (multi_oo_simulate_rollback_shots(frame_idx));
```
Comment (notimaginative) [inline code/network/multi_obj.cpp:741]: A bit of a nitpick, but the extra parenthesis should be removed. clang tends to complain about this too so I'm a little surprised that clang-tidy let it through.

**Review by notimaginative — CHANGES_REQUESTED** [review]
Just the one minor thing, but otherwise it looks good to me.

File: code/network/multi_obj.cpp [hunk 2]
```diff
@@ -496,25 +508,30 @@ int multi_ship_record_find_frame(int client_frame, int time_elapsed)
 	for (int i = Oo_info.cur_frame_index - 2; i > -1; i--) {
 
 		// Check to see if the client's timestamp matches the recorded frames.
-		if ((Oo_info.timestamps[i] <= target_timestamp) && (Oo_info.timestamps[i + 1] > target_timestamp)) {
+		if ((Oo_info.timestamps[i] <= target_timestamp) && (Oo_info.timestamps[i + 1] >= target_timestamp)) {
```
Comment (Goober5000) [inline code/network/multi_obj.cpp:511]: This looks like it could benefit from a timestamp ordering check.  I'll add that to the `timestamp_since` PR.

Comment (Goober5000) [reply code/network/multi_obj.cpp:511]: Or a new PR.  See #4324.

File: code/network/multi_obj.cpp (hunk 1 above)
Comment (JohnAFernandez) [reply code/network/multi_obj.cpp:741]: My reasoning was always to help indicate that it is a bool assignment.  I'll refrain from now on.

**Review by notimaginative — APPROVED** [review]

**Comment by JohnAFernandez** [comment]
Literally waiting on one last safety test before merging

**Review by notimaginative — APPROVED** [review]

**Pull request merged**